scripts/ddr3lib.py

```diff
@@ -0,0 +1,134 @@
+import math
+import collections
+import payload_ddr3_pb2
+import logging
+
+Timing = payload_ddr3_pb2.Timing
+Opcode = payload_ddr3_pb2.Opcode
+Instr = payload_ddr3_pb2.Instr
+Payload = payload_ddr3_pb2.Payload
+
+def VerifyInstr(instr: Instr) -> bool:
+  if instr.HasField('mem'):
+    mem = instr.mem
+    if mem.opcode not in {Opcode.RD, Opcode.ACT, Opcode.PRE, Opcode.REF}:
+      return False
+    if not (0 < mem.timeslice < (1 << Instr.MemInstr.Bits.TIMESLICE)):
+      return False
+    if mem.rank != 0:  # TODO: Add multi-rank support.
+      return False
+    if mem.bank >= 1 << Instr.MemInstr.Bits.BANK:
+      return False
+    if mem.addr >= 1 << Instr.MemInstr.Bits.ADDR:
+      return False
+    return True
+  if instr.HasField('nop'):
+    nop = instr.nop
+    if nop.opcode != Opcode.NOP:
+      return False
+    if not (0 < nop.timeslice < (1 << Instr.NopInstr.Bits.TIMESLICE)):
+      return False
+    return True
+  if instr.HasField('jmp'):
+    jmp = instr.jmp
+    if jmp.opcode != Opcode.JMP:
+      return False
+    if not (0 < jmp.offset < (1 << Instr.JmpInstr.Bits.OFFSET)):
```

Comment:
Is it worth verifying the jump target being a valid instruction?

Reply:
Yup, we do that in main(). Here we are just checking for SIGILLs.

Reply:
On second thought, that's a good idea. Done in the follow-up commit.